src/main/perl/Configuration.pm

```diff
@@ -464,12 +464,24 @@ sub elementExists
     return $ex;
 }
 
+# Handle failures. Stores the error message and
+# returns undef. All failures should use 'return $self->fail("message");'.
+# No error logging should occur in this module.
+sub fail
+{
+    my ($self, @messages) = @_;
+    $self->{fail} = join('', @messages);
```

Comment:
After the problem in the reporter, I'd suggest that this function checks that there is no `undef` in `@message` to avoid potential problems.